### PR TITLE
Fix/wp filter helper options

### PR DESCRIPTION
**Pull request by manwithtwowatches**
https://www.openproject.org/work_packages/7472
(see comment #36)

Options can be passed to the path helpers which is necessary for the atom links.

Updated the links on the roadmap view to use the new url helpers.

Wrote some specs.

### CODE REVIEW

**Comment by hschink** [comment]
@manwithtwowatches I noticed when clicking on the links on the project page that the global work package page is called and that the filters are not set correctly. For instance, when I click on _Assigned to me_ than the global work package page is shown with the standard filter (status) is set.

File: app/views/versions/_work_package_counts.html.erb [hunk 1]
```diff
@@ -53,12 +53,8 @@ See doc/COPYRIGHT.rdoc for more details.
...
           <tr>
             <td width="130px" align="right" >
-                <%= link_to h(count[:group]), {controller: :work_packages,
-                                            action: :index,
-                                            project_id: version.project,
-                                            set_filter: 1,
-                                            status_id: '*',
-                                            fixed_version_id: version}.merge("#{criteria}_id".to_sym => count[:group]) %>
+                <%= link_to h(count[:group]),
+                            project_version_property_path(version, "#{criteria}_id", count[:group].id) %>
```
Comment (hschink) [inline app/views/versions/_work_package_counts.html.erb:57]: _Author_ and _assignee_ are not properly passed to the work package view. The values for these filters are missing in the filter list.

**Comment by hschink** [comment]
@manwithtwowatches sorry, I mixed up the my page with the my project page! Nevertheless, it seems that the atom feeds on the project overview do not preserve the project scope.

File: app/views/versions/_work_package_counts.html.erb (hunk 1 above)
Comment (manwithtwowatches) [inline app/views/versions/_work_package_counts.html.erb:57]: @hschink you will need to show me this as it works ok for me.

Comment (manwithtwowatches) [inline app/views/versions/_work_package_counts.html.erb:57]: For the record this is unrelated to this PR. It's happening because the selected user isn't in the list of available users for the filter in the work packages table because they are not in that project anymore.

Comment (hschink) [inline app/views/versions/_work_package_counts.html.erb:57]: @manwithtwowatches is right!
